### PR TITLE
Api provides user to request items

### PR DESCRIPTION
**Pull request by pgaunitz**
### User Story
```
As a resident
In order to request items from someone shopping
I would like to contact and request items from that user
```
## Changes proposed in this pull request:
* Adding pong model
* Adding pongs controller
* Adding create action to pong controller
## What I have learned working on this feature:
* How to implement error handlers for any failure
* How to test for all possible sad paths

### CODE REVIEW

File: spec/requests/user_can_create_pong_spec.rb [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    it 'returns a 401 response status' do
+      expect(response).to have_http_status 401
+    end
+    it 'returns success message' do
+      expect(
+        response_json['errors'].first
+      ).to eq 'You need to sign in or sign up before continuing.'
+    end
+  end
+end
```
Comment (johanbounce) [inline spec/requests/user_can_create_pong_spec.rb:237]: BEST FILE EVER

File: spec/models/pong_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  describe 'Database table' do
+    it { is_expected.to have_db_column :item1 }
+    it { is_expected.to have_db_column :item2 }
+    it { is_expected.to have_db_column :item3 }
+    it { is_expected.to have_db_column :active }
+    it { is_expected.to belong_to :user}
+    it { is_expected.to belong_to :ping}
+  end
+  describe 'Validations' do
```
Comment (kaylawoodbury) [inline spec/models/pong_spec.rb:18]: can we add a line here to keep the formatting consistent

**Review by kfrostare — APPROVED** [review]
Looks like you guys did a HUGE job, awesome /Kf

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -12,6 +12,7 @@
     it { is_expected.to have_db_column :tokens }
     it { is_expected.to have_db_column :role}
     it { is_expected.to have_many :pings }
+    it { is_expected.to have_many :pongs }
   end
   describe 'Validations' do
```
Comment (kaylawoodbury) [inline spec/models/user_spec.rb:18]: can we add a ling here to keep the formatting consistent

**Review by johanbounce — APPROVED** [review]

**Review by kaylawoodbury — APPROVED** [review]
Wow, that is quite a lot of code added! There were few a minor formatting items but otherwise I thought this PR looked pretty good. I know you mentioned refactoring with Thomas which will help the readability of the logic in the controller, so I APPROVE!

**Pull request merged**